util/pkg/vfs/azureblob.go

```diff
@@ -100,43 +104,40 @@ func (p *AzureBlobPath) Join(relativePath ...string) Path {
 
 // ReadFile returns the content of the blob.
 func (p *AzureBlobPath) ReadFile(ctx context.Context) ([]byte, error) {
-	var b bytes.Buffer
-	_, err := p.WriteTo(&b)
-	if err != nil {
-		return nil, err
-	}
-	return b.Bytes(), nil
-}
-
-// WriteTo writes the content of the blob to the writer.
-func (p *AzureBlobPath) WriteTo(w io.Writer) (n int64, err error) {
-	ctx := context.TODO()
+	klog.Infof("Reading: %s - %s", p.container, p.key)
 
 	client, err := p.getClient(ctx)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
-	cURL, err := client.newContainerURL(p.container)
+	get, err := client.DownloadStream(ctx, p.container, p.key, nil)
 	if err != nil {
-		return 0, err
+		var azErr *azcore.ResponseError
+		if errors.As(err, &azErr) {
+			if azErr.ErrorCode == "ContainerNotFound" || azErr.ErrorCode == "BlobNotFound" {
+				return nil, os.ErrNotExist
```

Comment:
can we use the `bloberror` package's HasCode function and these error code constants? Same with the other error checking in this file

https://github.com/Azure/azure-sdk-for-go/blob/ab4bfbc61843da34541746e017ab7aa2141d6461/sdk/storage/azblob/bloberror/error_codes.go#L66

Reply:
Niiice! 😄